pkg/cbpf/alu_instructions_test.go

```diff
@@ -0,0 +1,326 @@
+			instruction:          Mod(X),
+			wantInstructionClass: int32(pb.InsClass_InsClassAlu),
+			wantSrc:              int32(pb.SrcOperand_RegSrc),
+			wantOperationCode:    int32(pb.AluOperationCode_AluMod),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                int32(X),
+		},
+		{
+			testName:             "Mod with Int as Source",
+			instruction:          Mod(testK),
+			wantInstructionClass: int32(pb.InsClass_InsClassAlu),
+			wantSrc:              int32(pb.SrcOperand_Immediate),
+			wantOperationCode:    int32(pb.AluOperationCode_AluMod),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                testK,
+		},
+		{
+			testName:             "Xor with Register as Source",
+			instruction:          Xor(X),
+			wantInstructionClass: int32(pb.InsClass_InsClassAlu),
+			wantSrc:              int32(pb.SrcOperand_RegSrc),
+			wantOperationCode:    int32(pb.AluOperationCode_AluXor),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                int32(X),
+		},
+		{
+			testName:             "Xor with Int as Source",
+			instruction:          Xor(testK),
+			wantInstructionClass: int32(pb.InsClass_InsClassAlu),
+			wantSrc:              int32(pb.SrcOperand_Immediate),
+			wantOperationCode:    int32(pb.AluOperationCode_AluXor),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                testK,
+		},
+		{
+			testName:             "Return with Register as Source",
+			instruction:          Ret(A),
+			wantInstructionClass: int32(pb.InsClass_InsClassRet),
+			wantSrc:              int32(pb.SrcOperand_RegSrc),
+			wantOperationCode:    int32(0x00),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                int32(A),
+		},
+		{
+			testName:             "Return with Int as Source",
+			instruction:          Ret(testK),
+			wantInstructionClass: int32(pb.InsClass_InsClassRet),
+			wantSrc:              int32(pb.SrcOperand_Immediate),
+			wantOperationCode:    int32(0x00),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                testK,
+		},
+		{
+			testName:             "Misc TAX",
+			instruction:          Misc(A),
+			wantInstructionClass: int32(pb.InsClass_InsClassMisc),
+			wantSrc:              int32(0x00),
+			wantOperationCode:    int32(0x00),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                int32(A),
+		},
+		{
+			testName:             "Misc TXA",
+			instruction:          Misc(X),
+			wantInstructionClass: int32(pb.InsClass_InsClassMisc),
+			wantSrc:              int32(0x00),
+			wantOperationCode:    int32(0x80),
+			wantJmpTrue:          0,
+			wantJmpFalse:         0,
+			wantK:                int32(X),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.testName, func(t *testing.T) {
+			instruction := tc.instruction
+			t.Logf("Running test case %s", tc.testName)
+
+			if (instruction.Opcode & 0x07) != tc.wantInstructionClass {
```

Comment:
consider putting the & 0x07 and each other "magic number" in a variable with a name for readability purposes, that way it would be more clear what each bit wise and operation is doing